src/server/journal/streamer.cc

```diff
@@ -378,8 +378,11 @@ void RestoreStreamer::Run() {
 
   PrimeTable::Cursor cursor;
   uint64_t last_yield = 0;
-  PrimeTable* pt = &db_array_[0]->prime;
 
+  // Explicitly copy table smart pointer to keep reference count up (flushall drops it)
+  boost::intrusive_ptr<DbTable> table = db_array_.front();
+  PrimeTable* pt = &table->prime;
+  ExpireTable& expire_table = table->expire;
```

Comment:
I remember we had the same with flush slots fiber, but I didn't think about checking other places as well 😞